extensions/blocks/simple-payments/simple-payments.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Pay with PayPal block (aka Simple Payments).
+ *
+ * @since 9.0.0
+ *
+ * @package Jetpack
+ */
+
+namespace Automattic\Jetpack\Extensions\SimplePayments;
+
+use Jetpack_Simple_Payments;
+
+const FEATURE_NAME = 'simple-payments';
+const BLOCK_NAME   = 'jetpack/' . FEATURE_NAME;
+
+/**
+ * Registers the block for use in Gutenberg
+ * This is done via an action so that we can disable
+ * registration if we need to.
+ */
+function register_block() {
+	jetpack_register_block(
+		BLOCK_NAME,
+		array(
+			'render_callback' => __NAMESPACE__ . '\render_block',
+			'plan_check'      => true,
+		)
+	);
+}
+add_action( 'init', __NAMESPACE__ . '\register_block' );
+
+/**
+ * Pay with PayPal block dynamic rendering.
+ *
+ * @param array  $attr    Array containing the block attributes.
+ * @param string $content String containing the block content.
+ *
+ * @return string
+ */
+function render_block( $attr, $content ) {
+	// Do nothing if block content is a `simple-payment` shortcode.
+	if ( preg_match( '/\[simple-payment(.*)]/', $content ) ) {
+		return $content;
+	}
+
+	$simple_payments = Jetpack_Simple_Payments::getInstance();
+	$simple_payments->enqueue_frontend_assets();
+
+	// Keep content as-is if rendered in other contexts than frontend (i.e. feed, emails, API, etc.).
+	if ( ! jetpack_is_frontend() ) {
+		return $content;
+	}
+
+	// Augment block UI with a PayPal button if rendered on the frontend.
+	$product_id  = $attr['productId'];
+	$dom_id      = uniqid( "jetpack-simple-payments-{$product_id}_", true );
```

Comment:
Should we use `wp_unique_id()` here?

Reply:
That sounds good. I used `uniqid` because it's what we're using in the [shortcode](https://github.com/Automattic/jetpack/blob/b8648d7ca8f3d9209f3b6f95fe00a2e760af7876/modules/simple-payments/simple-payments.php#L139), but `wp_unique_id` seems enough for this use case. Updated in d398969.